pkg/epp/handlers/response_test.go

```diff
@@ -155,3 +155,91 @@ func TestHandleStreamedResponseBody(t *testing.T) {
 		})
 	}
 }
+
+func TestParseRespForUsage(t *testing.T) {
+	ctx := logutil.NewTestLoggerIntoContext(context.Background())
+	tests := []struct {
+		name     string
+		response string
+		want     ResponseBody
+	}{
+		{
+			name:     "with usage",
+			response: `data: {"id":"test","usage":{"prompt_tokens":5,"total_tokens":10,"completion_tokens":5}}`,
+			want: ResponseBody{
+				Usage: Usage{
+					PromptTokens:     5,
+					TotalTokens:      10,
+					CompletionTokens: 5,
+				},
+			},
+		},
+		{
+			name:     "without usage",
+			response: `data: {"id":"test","choices":[{"text":"hello"},"usage":null]}`,
+			want:     ResponseBody{},
+		},
+		{
+			name: "multiple lines with usage",
+			response: `data: {"id":"test","choices":[{"text":"hello"},"usage":null]}
+data: {"id":"test","usage":{"prompt_tokens":3,"total_tokens":8,"completion_tokens":5}}
+`,
+			want: ResponseBody{
+				Usage: Usage{
+					PromptTokens:     3,
+					TotalTokens:      8,
+					CompletionTokens: 5,
+				},
+			},
+		},
+		{
+			name: "with [DONE] marker",
+			response: `data: {"id":"test","choices":[{"text":"hello"},"usage":null]}
+data: {"id":"test","usage":{"prompt_tokens":5,"total_tokens":10,"completion_tokens":5}}
+data: [DONE]`,
+			want: ResponseBody{
+				Usage: Usage{
+					PromptTokens:     5,
+					TotalTokens:      10,
+					CompletionTokens: 5,
+				},
+			},
+		},
+		{
+			name:     "with [DONE] marker only",
+			response: `data: [DONE]`,
+			want:     ResponseBody{},
+		},
+		{
+			name: "contains empty data lines",
+			response: `data: 
+data: {"id":"test","usage":{"prompt_tokens":2,"total_tokens":6,"completion_tokens":4}}`,
```

Comment:
You dup'd `data:` here causing CI to fail: https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_gateway-api-inference-extension/1432/pull-gateway-api-inference-extension-verify-main/1959612700963115008